android/walletlib/src/main/java/com/solana/mobilewalletadapter/walletlib/scenario/LocalScenario.java

```diff
@@ -0,0 +1,403 @@
+    private static final String TAG = LocalScenario.class.getSimpleName();
+
+    @NonNull
+    public byte[] associationPublicKey;
+
+    @NonNull
+    protected final MobileWalletAdapterConfig mMobileWalletAdapterConfig;
+    @NonNull
+    protected final Looper mIoLooper;
+    @NonNull
+    protected final Handler mIoHandler;
+    @NonNull
+    protected final Scenario.Callbacks mCallbacks;
+    @NonNull
+    protected final AuthRepository mAuthRepository;
+
+    private final Object mLock = new Object();
+    @Nullable
+    @GuardedBy("mLock")
+    private AuthRecord mActiveAuthorization = null;
+
+    private final PowerConfigProvider mPowerManager;
+
+    @Nullable
+    private ScheduledFuture<?> mNoConnectionTimeoutHandler;
+    private final ScheduledExecutorService mTimeoutExecutorService =
+            Executors.newSingleThreadScheduledExecutor();
+
+    protected LocalScenario(@NonNull Context context,
+                            @NonNull MobileWalletAdapterConfig mobileWalletAdapterConfig,
+                            @NonNull AuthIssuerConfig authIssuerConfig,
+                            @NonNull Scenario.Callbacks callbacks,
+                            @NonNull byte[] associationPublicKey) {
+        this(context, mobileWalletAdapterConfig, authIssuerConfig, callbacks,
+                associationPublicKey, new DevicePowerConfigProvider(context));
+    }
+
+    /*package*/ LocalScenario(@NonNull Context context,
+                              @NonNull MobileWalletAdapterConfig mobileWalletAdapterConfig,
+                              @NonNull AuthIssuerConfig authIssuerConfig,
+                              @NonNull Scenario.Callbacks callbacks,
+                              @NonNull byte[] associationPublicKey,
+                              @NonNull PowerConfigProvider powerConfigProvider) {
+        mCallbacks = callbacks;
+        mMobileWalletAdapterConfig = mobileWalletAdapterConfig;
+        this.associationPublicKey = associationPublicKey;
+
+        final LooperThread t = new LooperThread();
+        t.start();
+        mIoLooper = t.getLooper(); // blocks until Looper is available
+        mIoHandler = new Handler(mIoLooper);
+
+        mAuthRepository = new AuthRepositoryImpl(context, authIssuerConfig);
+
+        this.mPowerManager = powerConfigProvider;
+    }
+
+    @Override
+    public byte[] getAssociationPublicKey() {
+        return associationPublicKey;
+    }
+
+    @Override
+    protected void finalize() {
+        mIoLooper.quitSafely();
+    }
+
+    public MessageReceiver createMessageReceiver() {
+        return new MobileWalletAdapterSession(
+                this,
+                new MobileWalletAdapterServer(mMobileWalletAdapterConfig, mIoLooper, mMethodHandlers),
+                mSessionStateCallbacks);
+    }
+
+    @Override
+    public void start() {
+        startNoConnectionTimer();
+    }
+
+    @Override
+    public abstract void close();
+
+    public Long getNoConnectionTimeout() {
```

Comment:
nope good catch!

Reply:
Super minor, but no need to use a boxed type `Long` here; primitive `long` will do. I suspect that this would be optimized away at build time, but 🤷 .

Reply:
dang I actually forgot about that distinction in Java. Kotlin made my java skills rusty lol